content/en/apps/guides/data/analytics/building-dbt-models.md

```diff
@@ -0,0 +1,340 @@
+
+
+### Testing models and dashboards
+
+It is highly encouraged to write dbt [tests](https://docs.getdbt.com/docs/build/data-tests) for application specific models to ensure that they are accurate and to avoid releasing broken models. Examples can be found in [CHT Pipeline](https://github.com/medic/cht-pipeline/tree/main/test).
+
+
+## Base Models
+
+All tables contain a `uuid` which is the primary key for the table; it is also the `_id` from the source CouchDB document.
+
+{{< figure src="cht-pipeline-er.png" link="cht-pipeline-er.png" class=" center col-16 col-lg-12" >}}
+
+### `couchdb`
+All documents are stored in the `couchdb` table; downstream models move fields from the document into fields and index them.
+This table can be used to get any field from a CouchDB document; however, it is recommended that the columns from downstream models be used instead for performance reasons.
+|Field|Description|
+|--|--|
+|`uuid`|CouchDB's unique identifier of the record|
+|`type`|The general type of the document, see below|
+|`doc`| JSON of the source document|
+
+### `data_record`
+All form responses are stored in the `data_record` table; see more details [in the database schema conventions]({{< ref "core/overview/db-schema#reports" >}}).
+This table contains columns for the contact who made the report, the parent of that contact, the report data, and a copy of fields in `jsonb` format.
+It is not recommended to use the fields column directly, but instead to add one new model for each form, moving the relevant fields for that form into columns
+
+|Field|Description|
+|--|--|
+|`uuid`|Unique identifier of the record. Note this is both the primary key for this table, and a foreign key to the `couchdb` table.|
+|`contact_uuid`| uuid of the `contact` who made the report|
+|`contact_parent_uuid`| uuid of the parent of `contact` who submitted the form (at the date `reported`; contacts parent may have changed since then, this column will not)|
+|`reported`| the reported timestamp from the couchdb document, stored as a date|
+
+For SMS forms, there is also an `sms_form` table which contains the raw message and sender phone number (for SMS form response, `contact_uuid` may be `NULL`)
+The `contact_uuid` column contains a foreign key to the `contacts` table.
+
+### `contacts`
+See a description of contact documents in CouchDB [in the database schema conventions]({{< ref "core/overview/db-schema#contacts-persons-and-places" >}}).
+Every person and place is stored in the `contacts` table. Persons and patients are stored in their own tables, but because contact types are configurable, other contact types do not have their own tables by default.
+The contact hierarchy defines "is a" relationships between contact types; e.g., a patient is a person is a contact. This is modeled as one table per type, where the `uuid` is both the primary key for the child table and a foreign key to the parent table.
+
+|Field|Description|
```

Comment:
The [columns](https://github.com/medic/cht-pipeline/blob/b4ecea6ac3abddd2d49e66dbd30a87409eae91dd/models/contacts/contact.sql#L17-L18) have also changed slightly in this model.